imgee/storage.py

```diff
@@ -216,7 +222,7 @@ def resize_and_save(img, size, is_thumbnail=False):
     Get the original image from local disk cache, download it from S3 if it misses.
     Resize the image and save resized image on S3 and size details in db.
     """
-    src_path = download_frm_s3(img.name + img.extn)
+    src_path = download_from_s3(img.filename)
```

Comment:
Just noticed this is for download. Since `img.extn` is calculated, does this mean we're not actually saving the exact filename as stored in S3? That is a problem.